includes/wikia/DefaultSettings.php

```diff
@@ -1830,3 +1830,14 @@
  * Reject attempts to fall back to the MediaWiki session for authentication.
  */
 $wgRejectAuthenticationFallback = true;
+
+/**
+ * Return template type for given template from Template Classification Service
+ */
+$wgAutoloadClasses['ExternalTemplateTypesProvider'] = "{$IP}/includes/wikia/parser/ExternalTemplateTypesProvider.class.php";
```

Comment:
You may want to consider moving the code to a subdir (parser/templatetypes/ ) and setting up autoload and hooks in a single file referenced from here